akka-actor/src/main/scala/akka/actor/dungeon/FaultHandling.scala

```diff
@@ -244,6 +244,7 @@ private[akka] trait FaultHandling { this: ActorCell =>
       if (freshActor eq failedActor) setActorFields(freshActor, this, self) // If the creator returns the same instance, we need to restore our nulled out fields.
 
       freshActor.aroundPostRestart(cause)
+      checkReceiveTimeout() // user may have set a receive timeout in preStart which is called from postRestart
```

Comment:
I think the problem here is because the `restarting` happened after the `checkReceiveTimeoutIfNeeded(msg, timeoutBeforeReceive)` been invoked

Reply:
checkReceiveTimeout(false)

Reply:
We want to reschedule though or else the timeout does not happen.

Reply:
What if there already have ongoing scheduling?

Reply:
The the receive timeout is "per incarnation", when the old incarnation is crashed its receive timeout is cleared (line 125 and 126)